### PR TITLE
[prefs] *nix encoding priority

### PR DESCRIPTION
**Pull request by szlldm**
-unused "Indexing priority" removed (only from the Preference dialog)
-new encoding priority for *nixes (only lowering or normal)
 lowering works without privileges

PS: it would be reasonable to remove Playback priority and priority options above normal (imho it is placebo); only encoding priority reduction makes sense (in this case PRIORITY_ENCODING_NIX could have been reunified with PRIORITY_ENCODING)

### CODE REVIEW

File: avidemux/common/ADM_commonUI/DIA_prefs.cpp [hunk 1]
```diff
@@ -354,7 +360,20 @@ std::string currentSdlDriver=getSdlDriverName();
...
-
+#else
+        diaMenuEntry priorityEntries[] = {
+                     {2,      QT_TRANSLATE_NOOP("adm","Normal"),NULL}
+                    ,{3,      QT_TRANSLATE_NOOP("adm","Below normal"),NULL}
+                    ,{4,      QT_TRANSLATE_NOOP("adm","Low"),NULL}
+        };
+        diaElemMenu menuEncodePriority(&encodeNixPriority, QT_TRANSLATE_NOOP("adm","_Encoding priority:"), NB_ITEMS(priorityEntries), priorityEntries);
+        diaElemReadOnlyText encodePriorityText(NULL,
+                QT_TRANSLATE_NOOP("adm","After an encoding is finished, the restoration of the original priority is only possible with the proper privileges."),NULL);
```
Comment (eumagga0x2a) [inline avidemux/common/ADM_commonUI/DIA_prefs.cpp:371]: I would prefer a very clear hint that this is practically a one-way street affecting everything if I am not mistaken, not just encoding. If you agree, I'll change the text accordingly when merged.

**Comment by eumagga0x2a** [comment]
> it would be reasonable to remove Playback priority and priority options above normal (imho it is placebo)

It might work™ on Windows.

File: avidemux/common/ADM_commonUI/DIA_prefs.cpp (hunk 1 above)
Comment (szlldm) [reply avidemux/common/ADM_commonUI/DIA_prefs.cpp:371]: Ok.

**Comment by szlldm** [comment]
i have simpified the DIA_prefs code as well

File: avidemux_core/ADM_coreUI/src/DIA_encoding.cpp [hunk 2]
```diff
@@ -36,8 +36,9 @@ DIA_encodingBase::DIA_encodingBase(uint64_t duration)
 DIA_encodingBase::~DIA_encodingBase( )
 {
     ADM_info("DiaEncodingBase: Destroying\n");
-#ifdef _WIN32
-	setpriority(PRIO_PROCESS, 0, _originalPriority);
+#ifndef __HAIKU__
+    // try to restore, at most nothing will happen
```
[the remark this answers is not in the document]
Comment (szlldm) [reply avidemux_core/ADM_coreUI/src/DIA_encoding.cpp:40]: return value check is enough, errno is redundant in this case

**Comment by eumagga0x2a** [comment]
Thanks!

**Pull request merged**